h2o-core/src/main/java/hex/grid/filter/AnyMatchPermutationFilter.java

```diff
@@ -0,0 +1,44 @@
+package hex.grid.filter;
+
+import hex.Model;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class AnyMatchPermutationFilter<MP extends Model.Parameters> implements PermutationFilter<MP> {
+
+  private List<FilterFunction<MP>> _filterFunctions;
+
+  @SafeVarargs
+  public AnyMatchPermutationFilter(FilterFunction<MP>... filterFunctions) {
+    _filterFunctions = Arrays.asList(filterFunctions);
+  }
+
+  /**
+   *
+   * @param permutation instance of {@link Model.Parameters} that represents particular
+   *                   combination of hyper parameters during grid search
+   * @return false if permutation is considered to be skipped by any of filter functions
+   */
+  @Override
+  public boolean test(MP permutation) {
+    boolean evaluatePermutation = true;
+
+    for (FilterFunction<MP> fun : _filterFunctions) {
+      if (!fun.test(permutation)) {
+        evaluatePermutation = false;
+        break;
+      }
+    }
+    final boolean activateGlobalDecision = evaluatePermutation; // have to do this as lambda expression accept only effectively final variables
+
+    _filterFunctions.forEach(ff -> ff.activate(activateGlobalDecision, permutation));
```

Comment:
Why does `activate` come with parameter `activateGlobalDecision`? In the current implementation, it never does anything if `activateGlobalDecision = false`. What is activate function supposed to do when it is false. I don't understand the semantics.

Reply:
@michalkurka your words are proof for what I was saying here https://github.com/h2oai/h2o-3/pull/4040#discussion_r350850166

Reply:
okay, who can answer me then? @seb-h2o?

can we simplify this?

Reply:
@michalkurka I can "revert" to my original approach. When we activate based on `supervisor`'s decision, as it observes things globally and control behaviour of the children/filter_functions. But you need to come to an agreement with @seb-h2o